src/Microsoft.Data.SqlClient/add-ons/AzureKeyVaultProvider/AzureKeyVaultProviderTokenCredential.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider
+{
+    internal class AzureKeyVaultProviderTokenCredential : TokenCredential
+    {
+        private const string Bearer = "Bearer ";
+
+        private AuthenticationCallback Callback { get; set; }
+
+        private string Authority { get; set; }
+
+        private string Resource { get; set; }
+
+        internal AzureKeyVaultProviderTokenCredential(AuthenticationCallback authenticationCallback, string masterKeyPath)
+        {
+            Callback = authenticationCallback;
+            using (HttpClient httpClient = new HttpClient())
```

Comment:
Instantiating `HttpClient` can be expensive. I recommend declaring this once and using it both here and in the `KeyClient`, which you can do using the `KeyClientOptions` to set HTTP transport options, including which `HttpClient` to use.